src/Splitio-net-core/Services/Shared/Classes/ConfigService.cs

```diff
@@ -0,0 +1,99 @@
+            };
+        }
+
+        public SelfRefreshingConfig ReadInMemoryConfig(ConfigurationOptions config)
+        {
+            var baseConfig = ReadBaseConfig(config);
+
+            return new SelfRefreshingConfig
+            {
+                SdkVersion = baseConfig.SdkVersion,
+                SdkSpecVersion = baseConfig.SdkSpecVersion,
+                SdkMachineName = baseConfig.SdkMachineName,
+                SdkMachineIP = baseConfig.SdkMachineIP,
+                LabelsEnabled = baseConfig.LabelsEnabled,
+                BaseUrl = string.IsNullOrEmpty(config.Endpoint) ? "https://sdk.split.io" : config.Endpoint,
+                EventsBaseUrl = string.IsNullOrEmpty(config.EventsEndpoint) ? "https://events.split.io" : config.EventsEndpoint,
+                SplitsRefreshRate = config.FeaturesRefreshRate ?? 5,
+                SegmentRefreshRate = config.SegmentsRefreshRate ?? 60,
+                HttpConnectionTimeout = config.ConnectionTimeout ?? 15000,
+                HttpReadTimeout = config.ReadTimeout ?? 15000,                
+                RandomizeRefreshRates = config.RandomizeRefreshRates,ConcurrencyLevel = config.SplitsStorageConcurrencyLevel ?? 4,
+                TreatmentLogRefreshRate = config.ImpressionsRefreshRate ?? 30,
+                TreatmentLogSize = config.MaxImpressionsLogSize ?? 30000,
+                EventLogRefreshRate = config.EventsPushRate ?? 60,
+                EventLogSize = config.EventsQueueSize ?? 5000,
+                EventsFirstPushWindow = config.EventsFirstPushWindow ?? 10,
+                MaxCountCalls = config.MaxMetricsCountCallsBeforeFlush ?? 1000,
+                MaxTimeBetweenCalls = config.MetricsRefreshRate ?? 60,
+                NumberOfParalellSegmentTasks = config.NumberOfParalellSegmentTasks ?? 5,
+                StreamingEnabled = config.StreamingEnabled ?? true,
+                AuthRetryBackoffBase = GetMinimunAllowed(config.AuthRetryBackoffBase ?? 1, 1, "AuthRetryBackoffBase"),
+                StreamingReconnectBackoffBase = GetMinimunAllowed(config.StreamingReconnectBackoffBase ?? 1, 1, "StreamingReconnectBackoffBase"),
+                AuthServiceURL = string.IsNullOrEmpty(config.AuthServiceURL) ? "https://auth.split.io/api/auth" : config.AuthServiceURL,
+                StreamingServiceURL = string.IsNullOrEmpty(config.StreamingServiceURL) ? "https://streaming.split.io/event-stream" : config.StreamingServiceURL,
+                ImpressionMode = config.ImpressionMode ?? ImpressionModes.Optimized  
+            };
+        }
+
+        private int GetMinimunAllowed(int value, int minAllowed, string configName)
+        {
+            if (value < minAllowed)
+            {
+                _log.Warn($"{configName} minumum allowed value: {minAllowed}");
+
+                return minAllowed;
+            }
+
+            return value;
+        }
+    }
+
+    public enum ConfingTypes
+    {
+        InMemory,
```

Comment:
wouldn't be localhost a valid mode too?

Reply:
we don't have specific config for localhost